### ISSUE TITLE
v2 transforms failing to apply same transformation when multiple inputs passed in (Mac M1)

### ISSUE THREAD

**Issue by elyall** [pytorch/vision#9184]
### 🐛 Describe the bug

As described [here](https://pytorch.org/blog/extending-torchvisions-transforms-to-object-detection-segmentation-and-video-tasks/#the-new-transforms-api) in 2022 and exhibited in [a recent feature post](https://docs.pytorch.org/vision/0.23/auto_examples/transforms/plot_rotated_box_transforms.html#transforms-illustrations), `v2.transforms` should apply the same transformation to all inputs during a single call. However on my M1 Macbook when I pass in multiple inputs each input is transformed uniquely as if they are passed in as separate calls.

```python
import torch
from torchvision.transforms import v2

array = torch.rand(1, 2, 3)
rotator = v2.RandomHorizontalFlip(0.5)

for _ in range(100):
    assert torch.equal(*rotator((array, array.clone()))), (
        "arrays were transformed differently"
    )
print("Huh, no failure... am I going crazy?")
```

### Versions

PyTorch version: 2.8.0
Is debug build: False
CUDA used to build PyTorch: None
ROCM used to build PyTorch: N/A

OS: macOS 15.5 (arm64)
GCC version: Could not collect
Clang version: 17.0.0 (clang-1700.0.13.5)
CMake version: Could not collect
Libc version: N/A

Python version: 3.12.7 (main, Oct 16 2024, 07:12:08) [Clang 18.1.8 ] (64-bit runtime)
Python platform: macOS-15.5-arm64-arm-64bit
Is CUDA available: False
CUDA runtime version: No CUDA
CUDA_MODULE_LOADING set to: N/A
GPU models and configuration: No CUDA
Nvidia driver version: No CUDA
cuDNN version: No CUDA
Is XPU available: False
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True

CPU:
Apple M1 Pro

Versions of relevant libraries:
[pip3] mypy==1.17.1
[pip3] mypy_extensions==1.1.0
[pip3] numpy==2.2.6
[pip3] numpydoc==1.9.0
[pip3] pytorch-lightning==2.5.2
[pip3] torch==2.8.0
[pip3] torchmetrics==1.8.0
[pip3] torchvision==0.23.0
[conda] Could not collect

**Comment by abhi-glitchhg** [comment]
I can reproduce it on Google Collab as well

**Comment by NicolasHug** [comment]
When you pass a pure tensor to a transform, only the *first* one is treated as an image. The rest of the pure tensor are treated as non-input and are passed-through, i.e. not transformed. So in the snippet above, the first tensor is flipped with a probability of 0.5, and the second one is never flipped.

This behavior is described the the note here: https://docs.pytorch.org/vision/stable/auto_examples/transforms/plot_transforms_getting_started.html#what-do-i-pass-as-input

This was needed to avoid treating every tensor as an image. Labels, for example, are typically represented as tensors as well, and they're not images.

If you want to pass multiple images and treat them as such, you'll just need to wrap the tensors into ` tv_tensor.Image` tensor subclass:

```py
from torchvision.tv_tensors import Image
# ...
    assert torch.equal(*flipper((Image(array), Image(array.clone()))))
# ...

```
